web/services/AI/ai_model.py

```
@@ -0,0 +1,177 @@
+import re
+from typing import Union
+
+import google.generativeai as genai
+
+logger = logging.getLogger(__name__)
+
+api_key = os.getenv("GOOGLE_API_KEY")
+
+if not api_key:
+    # Defer hard failure; functions will return structured error objects instead.
+    print("⚠️  GOOGLE_API_KEY not found – AI correction disabled")
```

Ruff (0.8.2)

String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)